### ISSUE TITLE
need to deprecate charset 

### ISSUE THREAD

**Issue by cloneluke** [TypeStrong/ts-node#1976]
### Search Terms
charset
deprecation

charset
https://github.com/TypeStrong/ts-node/blob/97f9afd046b66a0fe05a7d76e7a32f94b872016f/src/index.ts
https://github.com/microsoft/TypeScript/issues/51909

charset is being deprecated by typescript

**Comment by cspotcode** [comment]
What problems do you anticipate will happen in the near future?  This can help us decide what code changes to make, if any.

**Comment by cspotcode** [comment]
Will that happen with ts-node?  The code you linked appears to be unrelated.

**Comment by cspotcode** [comment]
You're talking about a configuration option for TS, but the code you linked is not using that option.  The code you linked is something different.

**Comment by cspotcode** [comment]
I wonder if this is coming from @tsconfig/bases.

**Comment by cloneluke** [comment]
https://github.com/TypeStrong/ts-node/blob/97f9afd046b66a0fe05a7d76e7a32f94b872016f/src/index.ts#L1575

**Comment by cspotcode** [comment]
Yeah, that code seems unrelated to typescript's deprecated charset option.

**Comment by cloneluke** [comment]
it has the word charset in it?

**Comment by cspotcode** [comment]
That code is creating a data URI to be interpreted by node, not TS.  It is not setting TypeScript's deprecated charset option.

TypeScript 5 and ts-node are working together without that error, according to this example:
https://github.com/TypeStrong/ts-node-repros/tree/ts-5

This makes sense: in my example, I am not setting the deprecated charset option, so I don't get that error.

**Comment by corkupine** [comment]
@cloneluke - ran into the same thing, and removed `charset` from `compilerOptions` in `tsconfig.json`
